tests/test_tables.nim

```diff
@@ -57,3 +57,22 @@ block:
 
   doAssert [{"j":"a"}].toJson() ==
     """[{"j":"a"}]"""
+
+block: # issue 52
+  type Color = enum
+    Red = "red", Green = "green", Blue = "blue"
+  let s = """{
+    "red": 1,
+    "blue": 3
+  }"""
+  var v = s.fromJson(Table[Color, int])
+  doAssert v.len == 2
+  doAssert v[Red] == 1
+  doAssert v[Blue] == 3
+
+block: # issue 52
+  type Answer {.pure.} = enum
+    A, B, C
+  let a = {Answer.A: "aaaa", Answer.B: "bbb"}.toTable
+  doAssert $(a.toJson().fromJson(Table[Answer, string])) ==
+    """{A: "aaaa", B: "bbb"}"""
```

Comment:
I don't think this is valid json, it should be """{"A": "aaaa", "B": "bbb"}"""

Reply:
Oops, you are right, will fix that!

Reply:
it took a moment and some failed experiments to realize that the above was indeed correct. `"""{A: "aaaa", B: "bbb"}"""` is not supposed to be valid json, it is the conversion to string of nim object.

I have now added a line to the test that shows that conversion to json is correct.